src/helpers/mathjax.ts

```diff
@@ -0,0 +1,206 @@
+import { debounce } from 'lodash';
+import { isEmpty, memoize } from 'lodash/fp.js';
+import WeakMap from 'weak-map';
+import { assertWindow } from '../utils';
+
+declare global {
+  interface Window {
+    MathJax?: any;
+    __debugMathJax?: boolean;
+  }
+}
+
+const MATH_MARKER_BLOCK  = '\u200c\u200c\u200c'; // zero-width non-joiner
+const MATH_MARKER_INLINE = '\u200b\u200b\u200b'; // zero-width space
+
+const MATH_RENDERED_CLASS = 'math-rendered';
+const MATH_MARKED_CLASS = 'math-marked';
+const MATH_DATA_SELECTOR = `[data-math]:not(.${MATH_RENDERED_CLASS})`;
+const MATH_ML_SELECTOR   = `math:not(.${MATH_RENDERED_CLASS})`;
+const COMBINED_MATH_SELECTOR = `${MATH_DATA_SELECTOR}, ${MATH_ML_SELECTOR}`;
+const MATHJAX_CONFIG = {
+  extensions: [],
+  showProcessingMessages: false,
+  skipStartupTypeset: true,
+  styles: {
+    '#MathJax_MSIE_Frame': {
+      left: '', right: 0, visibility: 'hidden',
+    },
+    '#MathJax_Message': {
+      left: '', right: 0, visibility: 'hidden',
+    },
+  },
+  tex2jax: {
+    displayMath: [[MATH_MARKER_BLOCK, MATH_MARKER_BLOCK]],
+    inlineMath:  [[MATH_MARKER_INLINE, MATH_MARKER_INLINE]],
+  },
+};
+
+let mathJaxLoadingQueue: (() => void)[] = [];
+
+const findProcessedMath = (root: Element): Element[] => Array.from(root.querySelectorAll('.MathJax math'));
+const findUnprocessedMath = (root: Element): Element[] => {
+  const processedMath = findProcessedMath(root);
+  return Array.from(root.querySelectorAll('math')).filter((node) => processedMath.indexOf(node) === -1);;
+};
+
+const findLatexNodes = (root: Element): Element[] => {
+  const latexNodes: Element[] = [];
+  for (const node of Array.from(root.querySelectorAll(MATH_DATA_SELECTOR))) {
+    const formula = node.getAttribute('data-math');
+
+    // Set textContent once so that resolveOrWait calls don't
+    // undo a MathJax pass before rendered class is applied
+    if (!node.classList.contains(MATH_MARKED_CLASS)) {
+      node.textContent = (node.tagName.toLowerCase() === 'div')
+        ? `${MATH_MARKER_BLOCK}${formula}${MATH_MARKER_BLOCK}`
+        : `${MATH_MARKER_INLINE}${formula}${MATH_MARKER_INLINE}`;
+      node.classList.add('math-marked');
```

Comment:
This seems like a safe change that resolves a rendering issue mentioned in the comment, but probably needs scrutiny.